src/sdk/pynni/nni/smac_tuner/smac_tuner.py

```diff
@@ -129,23 +131,39 @@ def _main_cli(self):
 
         return optimizer
 
+    @rewrite_nas_space
     def update_search_space(self, search_space):
-        """TODO: this is urgly, we put all the initialization work in this method, because initialization relies
-        on search space, also because update_search_space is called at the beginning.
+        """
         NOTE: updating search space is not supported.
 
         Parameters
         ----------
-        search_space:
+        search_space: dict
             search space
         """
+
+        # TODO: this is ugly, we put all the initialization work in this method, because initialization relies
+        #         on search space, also because update_search_space is called at the beginning.
+
         if not self.update_ss_done:
-            self.categorical_dict = generate_scenario(search_space)
+            updated_search_space = dict()
+            self.fixed_value_set = dict()
+            for key, val in search_space.items():
+                try:
+                    if "_type" in val and ((val["_type"] == "randint" and val["_value"][0] + 1 == val["_value"][1])
```

Comment:
too ugly

Reply:
What's your suggestion then?

Reply:
please write this logic in `generate_pcs`, convert them to `categorical `.